internal/ffaac/files_saver_worker.go

```diff
@@ -0,0 +1,57 @@
+package ffaac
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/utilitywarehouse/finance-fulfilment-archive-api-cli/internal/pb/bfaa"
+)
+
+type fileSaverWorker struct {
+	faaClient bfaa.BillFulfilmentArchiveAPIClient
+	fileChan  <-chan string
+	basedir   string
+}
+
+func (f *fileSaverWorker) Run(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case fn, ok := <-f.fileChan:
+			if ok {
+				if err := f.sendFileToArchiveAPI(ctx, fn); err != nil {
+					logrus.Error(err)
+				}
+			} else {
+				return
+			}
+		}
+	}
+}
+
+func (f *fileSaverWorker) sendFileToArchiveAPI(ctx context.Context, fileName string) error {
+	logrus.Infof("Processing file %s", fileName)
+	file, err := os.Open(filepath.Join(f.basedir, fileName))
+	if err != nil {
+		return errors.Wrapf(err, "failed to open file %s", fileName)
```

Comment:
Go 1.3 is out, shall we try out the new error wrapping?

Reply:
Yep, good idea